chihiro_todo/main.js

```diff
@@ -1,69 +1,198 @@
+        }
+        task.status = newStatus;
+        appendTask(task);
+    }
 
-    doneArea .innerHTML = taskList.map((task, i) => {
+    const appendTask = function(task) {
+        newTaskElm = document.createElement('li');
+        const person = personList.find(person => person.id === task.personId);
+        taskPersonElm = document.createElement('span');
+        newTaskElm.setAttribute("id", "task_" + task.id);
+        newTaskElm.textContent = task.title;
+        taskPersonElm.textContent = person.name;
+        if (task.status === 0) {
+            todoArea.appendChild(newTaskElm);
+        }
+        if (task.status === 1) {
+            doingArea.appendChild(newTaskElm);
+        }
         if (task.status === 2) {
-            return `<li>${task.title}<button onclick="deleteTask(${i})">delete</button><div><button onclick="todoTask(${i})">todo</button><button onclick="doingTask(${i})">doing</button></li></div>`
+            doneArea.appendChild(newTaskElm);
+        }
+        
+        if (task.status !== 0) {
+            const todoButton = document.createElement('button');
+            todoButton.textContent = 'todo';
+            todoButton.addEventListener('click', function() {
+                changeStatus(task.id, 0);
+            })
+            newTaskElm.appendChild(todoButton);
+        }
+        if (task.status !== 1) {
+            const doingButton = document.createElement('button');
+            doingButton.textContent = 'doing';
+            doingButton.addEventListener('click', function() {
+                changeStatus(task.id, 1);
+            })
+            newTaskElm.appendChild(doingButton);
+        }
+        if (task.status !== 2) {
+            const doneButton = document.createElement('button');
+            doneButton.textContent = 'done';
+            doneButton.addEventListener('click', function() {
+                changeStatus(task.id, 2);
+            })
+            newTaskElm.appendChild(doneButton);
+        }
+
+        const deleteButton = document.createElement('button');
+        deleteButton.textContent = 'delete';
+        deleteButton.addEventListener('click', function() {
+            deleteTask(task.id);
+        })
+        newTaskElm.appendChild(deleteButton);
+        
+        newTaskElm.appendChild(taskPersonElm);
+    }
+    const defaultPerson = function() {
+        personList.forEach((person) => {
+            defaultPersonElm = document.createElement('li');
+            defaultPersonElm.setAttribute("id", "person_" + person.id);
+            defaultPersonElm.textContent = person.name;
+            personListArea.appendChild(defaultPersonElm);
+
+            deleteButton = document.createElement('button');
+            deleteButton.textContent = 'delete';
+            deleteButton.addEventListener('click', function() {
+                deletePerson(person.id);
+            })
+            defaultPersonElm.appendChild(deleteButton);
+        });
```

Comment:
@chihiro-kakizaki 
この`forEach`の中で`appendPerson`を呼んであげたら、処理をまとめられそうです！
```suggestion
        personList.forEach((person) => {
            appendPerson(person);
        });
```